### PR TITLE
Events controller & index+show views

### PR DESCRIPTION
**Pull request by Woulei**
issues #59 #76

### CODE REVIEW

File: spec/controllers/events_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
```
Comment (anikaSchwing) [inline spec/controllers/events_controller_spec.rb]: If this is empty, I think we should get rid of it

File: app/controllers/events_controller.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
+class EventsController < ApplicationController
+  before_action :set_event, only: [:show, :edit, :update]
+  before_action :authenticate_user!
+
+  def index
+    @events = Event.all
+    @public_events = @events.where(public: true)
```
Comment (arnoFleming) [inline app/controllers/events_controller.rb]: Fat models should hold this logic!

Rewrite it so you can call this:

```ruby
@public_events = Event.open_for_public
@incompany_events = Event.in_company
```

File: db/migrate/20170330091317_update_events_table.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class UpdateEventsTable < ActiveRecord::Migration[5.0]
+  change_table :events do |t|
+    t.rename :spaces, :spaces_available
```
Comment (arnoFleming) [inline db/migrate/20170330091317_update_events_table.rb]: YES

File: app/views/events/_incompany_events.html.erb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+      <th>#</th>
+      <th>Title</th>
+      <th>Location</th>
+      <th>Spaces Available</th>
+      <th>Date</th>
+      <th></th>
+    </tr>
+  </thead>
+  <tbody>
+    <% if @incompany_events.count > 0 %>
```
Comment (arnoFleming) [inline app/views/events/_incompany_events.html.erb]: Would also work: `<% if @incompany_events.count %>
`

File: app/views/events/_public_events.html.erb [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+      <th>#</th>
+      <th>Title</th>
+      <th>Location</th>
+      <th>Spaces Available</th>
+      <th>Date</th>
+      <th></th>
+    </tr>
+  </thead>
+  <tbody>
+    <% if @public_events != [] %>
```
Comment (arnoFleming) [inline app/views/events/_public_events.html.erb]: wut?

File: app/views/events/_public_events.html.erb [hunk 6]
```diff
@@ -0,0 +1,29 @@
+<h3>Public Events</h3>
+<table class="table table-striped">
```
Comment (arnoFleming) [inline app/views/events/_public_events.html.erb]: remove duplication

File: app/views/events/index.html.erb [hunk 7]
```diff
@@ -0,0 +1,6 @@
+<div class="container">
+  <h2>Events</h2>
+  <%= render 'public_events' %>
```
Comment (arnoFleming) [inline app/views/events/index.html.erb]: render 1 partial and use a local variable `:event` in there!

File: app/views/events/show.html.erb [hunk 8]
```diff
@@ -0,0 +1,8 @@
+<div class="container">
+  <h2><%= @event.title %></h2>
+  <strong>Location: </strong><%= @event.location %></br>
+  <strong>Spaces Available: </strong><%= @event.spaces_available %></br>
+  <strong>Date: </strong><%= @event.date %>
+  <br><br>
```
Comment (arnoFleming) [inline app/views/events/show.html.erb]: remove `br` and use CSS if you need padding or margin on your pages

File: spec/controllers/events_controller_spec.rb (hunk 1 above)
Comment (arnoFleming) [inline spec/controllers/events_controller_spec.rb]: +1

File: spec/factories/events.rb [hunk 9]
```diff
@@ -2,7 +2,8 @@
   factory :event do
     title "Factory Title"
     location "Factory Lane"
-    spaces 1
-    public false
+    spaces_available 35
+    public true
+    date 10.days.from_now
```
Comment (arnoFleming) [inline spec/factories/events.rb]: tell the workshop person to add it here as well, after merging this PR

File: spec/features/events/index_html_erb_spec.rb [hunk 10]
```diff
@@ -0,0 +1,30 @@
+require 'rails_helper'
+
+RSpec.describe 'events/index', type: :feature do
+  before { login_as user }
```
Comment (arnoFleming) [inline spec/features/events/index_html_erb_spec.rb]: Do I need to be logged in in this feature?

File: spec/helpers/events_helper_spec.rb [hunk 11]
```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
```
Comment (arnoFleming) [inline spec/helpers/events_helper_spec.rb]: please remove this file

File: spec/views/events/index_html_erb_spec.rb [hunk 12]
```diff
@@ -0,0 +1,4 @@
+require 'rails_helper'
+
+RSpec.describe "events/index.html.erb", type: :view do
+end
```
Comment (arnoFleming) [inline spec/views/events/index_html_erb_spec.rb]: add tests or remove

File: app/assets/javascripts/events.js [hunk 13]
```diff
@@ -0,0 +1,5 @@
+jQuery(document).ready(function($) {
+    $(".clickable-row").click(function() {
+        window.location = $(this).data("href");
```
Comment (arnoFleming) [inline app/assets/javascripts/events.js]: maybe you don't need this - clicking on actual html links is better testable

File: app/views/events/_incompany_events.html.erb (hunk 4 above)
Comment (Woulei) [inline app/views/events/_incompany_events.html.erb]: If I remove the `> 0` then the else statement doesn't work anymore.

File: spec/features/events/index_html_erb_spec.rb (hunk 10 above)
Comment (Woulei) [inline spec/features/events/index_html_erb_spec.rb]: Yes, this is in the admin scope!

File: app/assets/javascripts/events.js (hunk 13 above)
Comment (Woulei) [inline app/assets/javascripts/events.js]: Turns out that you can have an `a` element between `tbody` and `tr`. HTML won't allow it.
So, kept this in and made an invisible `a` around the name for testing.

File: app/views/events/_public_events.html.erb (hunk 6 above)
Comment (Woulei) [inline app/views/events/_public_events.html.erb]: Done

Comment (arnoFleming) [inline app/views/events/_public_events.html.erb]: thanks for actually replying! :D

File: app/views/events/_events_table.html.erb [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+      <th>#</th>
+      <th>Title</th>
+      <th>Location</th>
+      <th>Spaces Available</th>
+      <th>Date</th>
+      <th></th>
+    </tr>
+  </thead>
+  <tbody>
+    <% if events.count > 0 %>
```
Comment (arnoFleming) [inline app/views/events/_events_table.html.erb]: YES!

File: app/views/events/_incompany_events.html.erb [hunk 15]
```diff
@@ -0,0 +1,29 @@
+<h3>Incompany Events</h3>
```
Comment (arnoFleming) [inline app/views/events/_incompany_events.html.erb]: This partial isn't used anymore: Remove it!

File: app/views/events/_public_events.html.erb [hunk 16]
```diff
@@ -0,0 +1,29 @@
+<h3>Public Events</h3>
```
Comment (arnoFleming) [inline app/views/events/_public_events.html.erb]: This partial isn't used anymore: Remove it!

File: app/views/events/index.html.erb [hunk 17]
```diff
@@ -0,0 +1,8 @@
+<div class="container">
+  <h2>Events</h2>
+  <h3>Public Events</h3>
+  <%= render partial: "events_table", locals: { events: @public_events } %>
+  <hr>
+  <h3>InCompany Events
+  <%= render partial: "events_table", locals: { events: @incompany_events } %>
```
Comment (arnoFleming) [inline app/views/events/index.html.erb]: so pretty!

File: spec/features/events/index_html_erb_spec.rb (hunk 10 above)
Comment (arnoFleming) [inline spec/features/events/index_html_erb_spec.rb]: I cannot see that: You visit the `events_path`...

**Pull request merged**